src/platform_impl/macos/view.rs

```diff
@@ -79,6 +80,9 @@ pub(super) struct ViewState {
     /// True if the current key event should be forwarded
     /// to the application, even during IME
     forward_key_to_app: bool,
+
+    raw_characters: Option<String>,
```

Comment:
Odd, I stripped all this state. I found this and insert_text were causing duplicate characters, and insert_text was unnecessary. I wasn't able to reproduce any previous issues with the improved key_down handling.

Reply:
Going through the thread is a bit of an archaeological dig. It appears that @chrisduerr added it back in the last patch.. if I am getting the timeline right. I was getting duplicate characters too.. I'll poke around with that. I also prefer to avoid state if possible.

Reply:
Just fyi I wouldn't put too much faith in my path. I at least don't remember writing it at all and it was mostly an alternative suggestion to begin with.

Reply:
Sounds good.  I think I need to spend some more time debugging, as neither version seems to be working now.